zstacklib/zstacklib/utils/fileUploader.py

```
@@ -0,0 +1,289 @@
+
+    def upload_slice(self, entity, param, task):
+        # type: (Entity, UploadParam, UploadTask) -> None
+
+        boundary = get_boundary(entity)
+        if not boundary:
+            self._fail_task(task, 'unexpected post form')
+
+        try:
+            stream_body(entity, boundary, param, task)
+        except cherrypy.HTTPError as e:
+            raise cherrypy.HTTPError(e.status, e._message)
```

Ruff (0.13.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)